src/main/scala/fastq.scala

```diff
@@ -165,91 +165,91 @@ case object fastq {
       if(value.isEmpty)
         0
       else
-        value.foldLeft(0: BigDecimal){ (acc, v) => acc + (v: BigDecimal) } / (value.length)
+        value.foldLeft(0: BigDecimal){ _ + _ } / (value.length)
   }
 
   case object Quality {
 
     def fromPhred33(raw: String): Option[Quality] = {
 
       @annotation.tailrec
-      def rec(cs: String, acc: Seq[Int], errors: Boolean): Option[Quality] =
+      def rec(cs: String, acc: collection.mutable.Builder[Int,Vector[Int]], errors: Boolean): Option[Quality] =
         if(errors) { None } else {
 
-          if(cs.isEmpty) Some( Quality(acc) ) else {
+          if(cs.isEmpty) Some( Quality(acc.result) ) else {
 
             cs.head.toInt match {
-              case q if 33 <= q && q <= 126 => rec(cs.tail, acc :+ (q - 33), errors)
+              case q if 33 <= q && q <= 126 => rec(cs.tail, acc += (q - 33), errors)
               case _                        => rec(cs, acc, errors = true)
             }
           }
         }
 
-      rec(raw, Seq.empty, false)
+      val bldr = Vector.newBuilder[Int]; bldr.sizeHint(raw.length)
+
+      rec(raw, bldr, false)
     }
 
     val toPhred33: Int => Char = { i => (i + 33).toChar }
   }
 
-  class Id private[fastarious] (val value: String) extends AnyVal {
+  case class Id private[fastarious] (val value: String) extends AnyVal {
 
     def asString: String =
       s"@${value}"
+
+    def isEmpty: Boolean =
+      value.isEmpty
   }
 
   case object Id {
 
-    private val isValid: String => Boolean = { raw =>
-      (raw startsWith "@") &&
-      !(raw contains "\n")
-    }
-
-    def from(raw: String): Option[Id] = {
+    def from(raw: String): Id =
+      new Id(raw.filterNot(_ == '\n'))
```

Comment:
Are you sure you want to join several lines in one? Or probably you meant `takeWhile(_ != '\n')`?